src/sw/redis++/connection.cpp

```diff
@@ -93,6 +99,54 @@ auto ConnectionOptions::_split_path(const std::string &path) const
     return std::make_tuple(path, 0);
 }
 
+template <>
+void ConnectionOptions::_set_optional_key<bool>(const std::string &options_uri, const std::string &key, bool &field) const {
+    if (auto option = _extract_string_option(options_uri, key)) {
+        if (option != "true" && option != "false") {
```

Comment:
we cannot compare a `Optional<std::string>` and a c-style char array. so I think the code cannot compile

Reply:
in c++17 it compiles since std::optional overloads `operator==` with std::string ([see here](https://en.cppreference.com/w/cpp/utility/optional/operator_cmp)) and a `std::string`  can be implicitly constructed from a c-style array

Reply:
My bad... I don't know there's an `operator==` to compare `std::string` and `std::optional` before. Thanks for pointing it out!